libs/books/feature/src/lib/reading-list/reading-list.component.ts

```diff
@@ -10,9 +12,23 @@ import { getReadingList, removeFromReadingList } from '@tmo/books/data-access';
 export class ReadingListComponent {
   readingList$ = this.store.select(getReadingList);
 
-  constructor(private readonly store: Store) {}
+  constructor(
+    private readonly store: Store,
+    private _snackBar: MatSnackBar
+  ) { }
 
   removeFromReadingList(item) {
     this.store.dispatch(removeFromReadingList({ item }));
+    this.openSnackBar("Book removed", "Undo", item);
+  }
+
+  openSnackBar(message: string, action: string, book: Book) {
```

Comment:
You've got roughly the same code in two places now. Could you consolidate this into an effect, perhaps?

Reply:
updated